### PR TITLE
Fix undeclared variable error

### PR DESCRIPTION
**Pull request by roygbyte**
Fixes https://github.com/koreader/koreader/pull/8214#issuecomment-918856569

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/8220)

### CODE REVIEW

File: plugins/newsdownloader.koplugin/feed_view.lua [hunk 1]
```diff
@@ -63,10 +63,9 @@ function FeedView:getItem(id, feed, edit_feed_callback, delete_feed_callback)
...
-    local download_full_article = feed.download_full_article == nil
-        or feed.download_full_article
-    local include_images = not never_download_images
-        and feed.include_images
+    local download_full_article = feed.download_full_article ~= nil
+    local include_images = feed.include_images
+        or feed.include_images == nil
     local enable_filter = feed.enable_filter
         or feed.enable_filter == nil
     local filter_element = feed.filter_element
```
Comment (poire-z) [inline plugins/newsdownloader.koplugin/feed_view.lua:71]: Don't you want to also clean these to use use `thing = feed.thing ~= false` ?

Comment (poire-z) [reply plugins/newsdownloader.koplugin/feed_view.lua:71]: And is this right for filter_element that will be true when it is nil ? Shouldn't it be an element name/string - or is it really a boolean ? (haven't checked the code, just asking).

Comment (roygbyte) [reply plugins/newsdownloader.koplugin/feed_view.lua:71]: > Don't you want to also clean these to use use `thing = feed.thing ~= false` ?

Maybe? I honestly can't wrap my head around this logic at the moment. But the intended functionality is based on two possible situations.

a) the user has a an older version of `feed_config.lua`, (i.e.: from when they had to enter the values via a text input dialog) which may or may not have values for `include_images`, `enable_filter`, `download_full_article`, and `filter_element`.

b) the user has a "newer" version of `feed_config.lua`, (i.e.: where feeds are added via the new feed list menu) which will, by default, include values for `include_images`, `enable_filter`, `download_full_article`, and `filter_element`. 

Whether this little bit of logic accommodates those two situations I'm not sure. But it seems to work when I test manually! XD

> And is this right for filter_element that will be true when it is nil ? Shouldn't it be an element name/string - or is it really a boolean ? (haven't checked the code, just asking).

It'll be a string if the user supplies the css selector. Otherwise, it should be nil. Does that functionality accord with the logic of the code?  
https://github.com/koreader/koreader/blob/5b049a5a8232edb5013af7d3509095fecf4b1a0a/plugins/newsdownloader.koplugin/epubdownloadbackend.lua#L24-L50

Comment (poire-z) [reply plugins/newsdownloader.koplugin/feed_view.lua:71]: > It'll be a string if the user supplies the css selector. Otherwise, it should be nil. 

Your current:
```lua
local enable_filter = feed.enable_filter
        or feed.enable_filter == nil
```

is equivalent to:
```lua
local enable_filter
if feed.enable_filter then -- anything except false or nil
    enable_filter = feed.enable_filter
else -- so, feed.enable_filter was either false or nil
    if feed.enable_filter == nil then -- it was nil
        enable_filter = true -- so you make nil becomes true
    else -- it was false
        enable_filter = false
    end
end
```
(`feed.enable_filter == nil` evaluates to a boolean: `true` or `false`)

So, this should work for all your other boolean settings.
If I read you and that snippet well, enable_filter should be a string, or stay `nil` ; it shouldnt be `true` or `false`, at least it shouldn't be `true`, we don't know what will happen later with the code that look in the selector array.

So, probably just:
```
local enable_filter = feed.enable_filter -- let this stay nil if it was nil
```

Comment (roygbyte) [reply plugins/newsdownloader.koplugin/feed_view.lua:71]: Oh, I think I get it now. :) If the variable is not set, it defaults to `true`. Otherwise, the variable will remain unchanged as either `false` or some other value. 

> If I read you and that snippet well, enable_filter should be a string, or stay nil.

`enable_filter` will always be true or false. `filter_element` is what you're thinking of. `filter_element` is set to "" (empty string) for new feeds, which (now that I think of it) isn't reflected as something that should be excluded in the filter method.

``` lua 
if element then 
     table.insert(selectors, 1, element) 
end 
```
Should actually be...

``` lua 
if element and element ~= "" then 
     table.insert(selectors, 1, element) 
end 
```

> Don't you want to also clean these to use use thing = feed.thing ~= false ?

Long story short: I'll change `include_images`, `enable_filter`, `download_full_article` to that style. `filter_element` I'll change to:

``` lua
local filter_element = feed.filter_element
```
